custom_components/fellow/fellow_aiden/__init__.py

```
@@ -325,6 +349,89 @@ def get_brewer_id(self) -> str | None:
+        return preferred + remaining
+
+    async def _fetch_list_resource(
+        self, url: str, action: str
+    ) -> list[dict[str, Any]]:
+        """Fetch a list payload from the Fellow API."""
+        response = await self._request_with_reauth("get", url)
+        await self._ensure_success(response, action)
+
+        parsed = await self._parse_response(response)
+        if not isinstance(parsed, list):
+            raise Exception(f"Unexpected {action.lower()} payload: {parsed}")
```

Ruff (0.15.10)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)